servicetalk-data-protobuf-jersey/build.gradle

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+buildscript {
+  dependencies {
+    classpath "com.google.protobuf:protobuf-gradle-plugin:$protobufGradlePluginVersion"
+  }
+}
+
+apply plugin: "io.servicetalk.servicetalk-gradle-plugin-internal-library"
+apply plugin: "com.google.protobuf"
+
+dependencies {
+  implementation platform("org.glassfish.jersey:jersey-bom:$jerseyVersion")
+  testImplementation enforcedPlatform("org.glassfish.jersey:jersey-bom:$jerseyVersion")
+
+  testImplementation enforcedPlatform("org.junit:junit-bom:$junit5Version")
+
+  implementation project(":servicetalk-annotations")
+  implementation project(":servicetalk-buffer-netty")
+  implementation project(":servicetalk-concurrent-api-internal")
+  implementation project(":servicetalk-concurrent-internal")
+  implementation project(":servicetalk-data-protobuf")
+  implementation project(":servicetalk-http-api")
+  implementation project(":servicetalk-http-router-jersey")
+  implementation project(":servicetalk-http-router-jersey-internal")
+  implementation project(":servicetalk-http-utils")
+  implementation project(":servicetalk-transport-netty")
+  implementation "com.google.protobuf:protobuf-java:$protobufVersion"
+  implementation "com.google.code.findbugs:jsr305:$jsr305Version"
+  implementation "org.glassfish.jersey.core:jersey-server"
```

Comment:
This module exposes `jersey-common` and `jakarta.ws.rs-api` to the public API. We have to define them as `api` dependencies.
`servicetalk-data-jackson-jersey` has the same issue.

Reply:
I see `public class ServiceTalkProtobufSerializerFeature implements Feature` which is where `jakarta.ws.rs-api` is exposed, where is `jersey-common` exposed?

Reply:
Ah, I see `public final class ServiceTalkProtobufSerializerAutoDiscoverable implements AutoDiscoverable`

Reply:
however these types are linked via class path discovery so I'm not sure they are required to put in the public api ... would `runtime` be sufficient in this case? https://search.maven.org/artifact/io.servicetalk/servicetalk-data-jackson-jersey/0.42.8/jar

Reply:
`FeatureContext` and `MediaType` from `jakarta.ws.rs-api` are part of the public API -> it have to be an `api` dependency.
Not 100% sure about `implements AutoDiscoverable`. Something reminds me that a user had an issue when a parent abstract class was from another module defined as `implementation` instead of `api`. I would make both `api` just to be safe. 

jackson module has the same issue.

Reply:
alright I'll add them in both for now ... harder to remove them later though.